comp/trace/compression/impl-zstd/zstd.go

```diff
@@ -0,0 +1,36 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package implzstd implements the compression component interface
+package implzstd
+
+import (
+	"io"
+
+	compression "github.com/DataDog/datadog-agent/comp/trace/compression/def"
+
+	"github.com/DataDog/zstd"
+)
+
+const encoding = "zstd"
+
+type compressor struct{}
+
+// NewComponent creates a new compression component
+func NewComponent() compression.Component {
+	return &compressor{}
+}
+
+func (c *compressor) NewWriter(w io.Writer) (io.WriteCloser, error) {
+	return zstd.NewWriterLevel(w, zstd.BestSpeed), nil
```

Comment:
I'm wondering if we want to standardise on the compression level the agent sets. Currently Metrics sets it by default to `DefaultCompression` (5). But this can be configured by the user. I was going to do similar for logs.

Is there a reason `BestSpeed` has been chosen here? Maybe we should change metrics to do the same level.

Reply:
`BestSpeed` provided the highest throughput on the trace-agent, with just a slightly lower compression ratio than the default one. See the linked document for the full analysis: https://docs.google.com/document/d/1HiMsdz8UDQEI-2Ry-lycCfvuMuCow_Scon8JWhCi9qc/edit#heading=h.ei4su89bdi9f

Reply:
That's quite a boost!